lib/src/main/kotlin/WeatherSensorService/Internal/WeatherSensorReading.kt

```diff
@@ -0,0 +1,26 @@
+//
+// SpinDance
+// Weather Sensor Service
+//
+// Copyright © 2022 SpinDance. All rights reserved.
+//
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+data class WeatherSensorReading( 
+    override var temperature: Double,
+    override var humidity: Double,
+    override var pressure: Double,
+    override var time: LocalDateTime) : WeatherSensorReadingType {
```

Comment:
These should be immutable (and I think the compiler will enforce that once the interface's properties are made `val`s)